### DISCUSSION TITLE
Put config in rustdesk.exe

### DISCUSSION THREAD

**Discussion by anthonymontebrun** [rustdesk/rustdesk#2135] [Q&A] [answered]
Hello,

I followed your documentation to auto configure clients (https://rustdesk.com/docs/en/self-host/install/#put-config-in-rustdeskexe-file-name-windows-only).
It does not work correctly, here is the name of the .exe: rustdesk-host=DOMAIN.FR,key=XXXXX.exe

I have no characters not accepted by Windows.

Sincerely,

**Comment by pedjas** [comment]
Do you have RestDesk client installed besides that one renamed exe?

**Reply by anthonymontebrun** [reply]
Hello, 
I don't have RestDesk client installed besides.
But I had the client installed in the past (maybe traces in the registry?)

Sincerely,

**Reply by pedjas** [reply]
Unlikely. If you deinstall it then it does not make problems any more.

Check if you have firewall blocking connections to server.

Try telnet to IP address of server and port 2117. If ti connects then firewall is not blocking.

**Comment by kangaroo72** [comment]
This should work - this is my name: rustdesk-host=xxxxxx.cc,key=sdfkjsdfsdfnY9i8yJp9smzzLnax9TNX7O3NaNhkWE8=
(Params are changed). Once started, look if you see the key in the "about"-menu.

**Reply by pedjas** [reply]
If key is wrong RustDesk client would try to connect and would be refused by server and it would report that.
If it cannot connect at all, then it is not issue with key.

**Comment by pedjas** [comment]
There are logs in %appdata%\RustDesk. Check if there is some useful info there.

**Reply by shaqaruden** [reply]
I'm having issues too. I have configured my shortcut like so
`"C:\Program Files\RustDesk\RustDesk.exe" rustdesk-host=rustdesk.listech.on.ca,key=*pub key here*`

but I get 
```
[2022-11-23 16:36:22.653078 -05:00] ERROR [src\ui.rs:134] Wrong command: ["rustdesk-host=server,key=9D4wDyE4uJFtQ9hCZhCyW32Rz5tF41QvpBnYxocBNOc="]
```

**Reply by kangaroo72** [reply]
@shaqaruden I think your setup is wrong.
It's NOT allowed to have the file rustdesk.exe on you harddrive. For your own setup, it should be like this:
If the host ist wonderland.com and the key is 123456789abcdefghij your file should be:
rustdesk-host=wonderland.com,key=123456789abcdefghij
It's working - I also hat first problems, but it's not magic ;-)
Good luck,
Tom

**Reply by pedjas** [reply]
> I'm having issues too. I have configured my shortcut like so `"C:\Program Files\RustDesk\RustDesk.exe" rustdesk-host=rustdesk.listech.on.ca,key=*pub key here*`

1. Remove installed RustDesk
2 .Download standalone exe, not installation.
3. Extract standalone exe somewhere on disk (better out of C:\Program Files)
4. Rename exe to get this file name: rstdesk-host=rustdesk.listech.on.ca,key=<pubkey>.exe
5. Start newly named exe.

**Reply by pedjas** [reply]
> @shaqaruden I think your setup is wrong. It's NOT allowed to have the file rustdesk.exe on you harddrive. 

Of course it is allowed. If you run rustdesk.exe it works connecting to public hosts. That is actually the only way to have option to connect to public hosts besides self hosted one.

**Reply by shaqaruden** [reply]
ah ok I get it now, never seen an application have configuration based on the filename before

**Comment by anthonymontebrun** [comment]
Hello,

I have Key mismatch when i connect, i have change key many Time but i have = at the end.

**Reply by shaqaruden** [reply]
Mine had a `=` sign at the end too but also a `?` when I ran `cat id_ed25519.pub` I omitted the `?` and it worked for me
![image](https://user-images.githubusercontent.com/5414897/203788418-e75b2c3e-0922-4a49-abab-8f0678c29106.png)

**Comment by kangaroo72** [comment]
I also had a "=" at the end of the key.
I mentioned, it's not allowed to have the file on the harddisk, when you rename it of course.
My customers load the file named like this, and are able to connect.

rustdesk-host=mydomain.tld,key=eiu5pzFYlr5nY9i9yJp9smzzLnvx4XNX7O3SaNhkWS8=.exe

(personal data modified)

**Comment by kangaroo72** [comment, marked as answer]
I have red a info, in the docs, that it's possible, to re-generated the key, if you have problems with a key.
If so, please do so. That's [here](https://rustdesk.com/docs/en/self-host/install/).

**Comment by WeiYea** [comment]
Modify "libs hbb_common src config. rs",

Change "RENDEZVOUS_SERVERS" to your address

**Reply by pedjas** [reply]
This should be user configurable, not changed in code and compiled.

**Reply by WeiYea** [reply]
In fact, as a user, I do not expect that the software needs to be configured with too many parameters. For users, the simpler the better. The configuration is completed in the code, which is easier for users to use.

**Reply by pedjas** [reply]
> In fact, as a user, I do not expect that the software needs to be configured with too many parameters. For users, the simpler the better. The configuration is completed in the code, which is easier for users to use.

Once they add option that host is part of ID, then there would not be need to configure client to connect to specific host as it would read it from id. As I understood that is work in progress.

**Comment by SynergIT-fr** [comment]
hello, 
I'm late to respond but it might be helpful.
I followed this how to and builed my own exe with my configuration information.
https://rustdesk.com/docs/en/dev/build/all/

**Comment by rustdesk** [comment]
This is an ad, custom client generator with config built-in and signed. https://twitter.com/rustdesk/status/1767588052325572682

**Comment by Andre0be** [comment]
A custom client generator for free is a good idea. Everything else does not work.
